packages/react-i18n/src/utilities/translate.tsx

```diff
@@ -12,11 +12,14 @@ import {
 } from '../types';
 import {MissingTranslationError, MissingReplacementError} from '../errors';
 
-const REPLACE_REGEX = /{([^}]*)}/g;
+import {DEFAULT_INTERPOLATION} from './interpolate';
+
 const MISSING_TRANSLATION = Symbol('Missing translation');
 const PLURALIZATION_KEY_NAME = 'count';
 const SEPARATOR = '.';
 
+const isString = (value: any): value is string => typeof value === 'string';
```

Comment:
Instead of having single-use variables like:

```ts
const isAllStrings = pieces.every((piece) => typeof piece === 'string');
if (isAllStrings) //...
```

Clear functional code is enough:

```ts
if (pieces.every(isString)) //...
```